Gems/Atom/RPI/Code/Source/RPI.Public/WindowContext.cpp

```diff
@@ -59,24 +59,51 @@ namespace AZ
             DestroySwapChain();
         }
 
-        const RHI::AttachmentId& WindowContext::GetSwapChainAttachmentId() const
-        {
-            return m_swapChain->GetAttachmentId();
+        const RHI::AttachmentId& WindowContext::GetSwapChainAttachmentId(SwapChainMode swapChainMode) const
+        {   
+            return GetSwapChain(swapChainMode)->GetAttachmentId();
         }
 
-        const RHI::Ptr<RHI::SwapChain>& WindowContext::GetSwapChain() const
+        const RHI::Ptr<RHI::SwapChain>& WindowContext::GetSwapChain(SwapChainMode swapChainMode) const
         {
-            return m_swapChain;
+            switch (swapChainMode)
+            {
+                case SwapChainMode::Default:
+                    return m_swapChain;
+                case SwapChainMode::XrLeft:
+                case SwapChainMode::XrRight:
+                    return m_xrSwapChains[static_cast<uint32_t>(swapChainMode)];
```

Comment:
If you change `Default` to be 0 in the enum, careful with using `swapChainMode` as index. Repeated in a few places.